src/rebuild/scan.c

```diff
@@ -366,11 +366,14 @@ rebuild_object_insert(struct rebuild_tgt_pool_tracker *rpt,
 		 * reclaim is not being scheduled in the previous failure reintegration,
 		 * so let's ignore duplicate shards(DER_EXIST) in this case.
 		 */
-		if (rpt->rt_rebuild_op == RB_OP_REINT || rpt->rt_rebuild_op == RB_OP_EXTEND) {
+		if (rpt->rt_rebuild_op == RB_OP_REINT || rpt->rt_rebuild_op == RB_OP_EXTEND)
 			D_DEBUG(DB_REBUILD, DF_UUID" found duplicate "DF_UOID" %d\n",
 				DP_UUID(co_uuid), DP_UOID(oid), tgt_id);
-			rc = 0;
-		}
+		else
+			/* Since it is harmless, let's skip duplicate obj for other cases. */
+			D_WARN(DF_UUID" found duplicate "DF_UOID" %d\n",
+			       DP_UUID(co_uuid), DP_UOID(oid), tgt_id);
```

Comment:
just confirm is it possibly caused by retried rebuild? or other case?

Reply:
Along those lines, did you figure out why it's getting a duplicate and that it is indeed an exact duplicate (no shard or pad difference)?

This seems like an iterator bug otherwise that should be fixed instead of adding a hack to work around it.

Reply:
> just confirm is it possibly caused by retried rebuild? or other case?

not sure, I can not reproduce the issue locally.

Reply:
> Along those lines, did you figure out why it's getting a duplicate and that it is indeed an exact duplicate (no shard or pad difference)?
> 
> This seems like an iterator bug otherwise that should be fixed instead of adding a hack to work around it.

yeah, that is totally possible, that is why I leave a warning here. My suspicion is that this is related with the vos_discard(), but the code looks fine to me.  And unfortunately I can not reproduce the issue here, so just leave a warning to make rebuild can pass for the moment.